tests/yam/migration/setup_upgrade_env.pm

```diff
@@ -22,14 +22,19 @@ sub run {
             get_var('SCC_ADDONS')));
 
     # Save the original value of the variables in order to restore it later if needed
-    set_var('VERSION_ENV', get_var('VERSION'));
-    set_var('SCC_ADDONS_ENV', get_var('SCC_ADDONS'));
+    foreach my $var (qw(AGAMA SCC_ADDONS VERSION)) {
+        set_var($var . "_ENV", get_var($var)) if (get_var($var));
+    }
 
     # Change variables to the other version that we want to migrate from/to
-    set_var('VERSION', $version);
-    record_info('VERSION', 'VERSION=' . get_var('VERSION'));
-    set_var('SCC_ADDONS', $scc_addons);
-    record_info('SCC_ADDONS', 'SCC_ADDONS=' . get_var('SCC_ADDONS'));
+    my @vars = ('AGAMA', 'SCC_ADDONS', 'VERSION');
+    my @val = (qw('0' $scc_addons $version));
+    for (my $i = 0; $i < @vars; $i++) {
+        if (get_var($vars[$i])) {
+            set_var($vars[$i], eval($val[$i]));
+            record_info($vars[$i], "$vars[$i]=" . eval($val[$i]));
+        }
+    }
```

Comment:
please, don't use index (in combination with eval) and if you create list in a way, be consistent, use always qw.